### PR TITLE
Tasks/9727

### PR DESCRIPTION
**Pull request by FuriousLlama**
# Fixes or Implements [AB#9727](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/9727)

* [x] Enhancement
* [ ] Bug
* [ ] Documentation

## Description

Added the Immz vuex store. If deferred, the store will try to query for labs until load state is updated

If you are reviewing this PR, please adhere to the guidelines as [documented](https://github.com/bcgov/healthgateway/wiki/prguidance).

## Testing

* [ ] Unit Tests Updated
* [ ] Functional Tests Updated
* [x] Not Required

### Steps to Reproduce

If this is a bug, please provide details on how to reproduce the code.

### UI Changes

YES

### Browsers Tested

* [x] Chrome
* [ ] Safari
* [ ] Edge
* [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant.  Include any specialized deployment steps here.

### CODE REVIEW

File: Apps/WebClient/src/ClientApp/src/store/modules/immunization/actions.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                    .then((immunizationRequestResult) => {
+                        if (
+                            immunizationRequestResult.resultStatus ===
+                            ResultType.Success
+                        ) {
+                            const results =
+                                immunizationRequestResult.resourcePayload;
+                            if (results.loadState.refreshInProgress) {
+                                logger.info("Immunizations load deferred");
+                                setTimeout(() => {
```
Comment (FuriousLlama) [inline Apps/WebClient/src/ClientApp/src/store/modules/immunization/actions.ts:44]: This is the section that does the magic on the VUEX side

File: Apps/WebClient/src/ClientApp/src/views/timeline.vue [hunk 2]
```diff
@@ -489,6 +502,54 @@ export default class TimelineView extends Vue {
...
+            });
+    }
+
+    private loadImmunizationEntries() {
+        if (this.immunizationLoadReady) {
+            this.immunizationLoadDeferred = false;
+            this.immunizationLoadReady = false;
+        }
+        // Add the immunization entries to the timeline list
+        for (let immunization of this.patientImmunizations) {
```
Comment (FuriousLlama) [inline Apps/WebClient/src/ClientApp/src/views/timeline.vue:544]: We should switch to this pattern to all other data as well. Request from Vuex for the load and have the vue reactivity update the timelineEntries array as it gets updated. That way the logic for populating is centralized and vuex is reponsable of maintaining state of the timeline

**Comment by FuriousLlama** [comment]
Changes regarding Comments are showing from the merge from dev. Safe to ignore in this PR

File: Apps/Database/src/Delegates/DBCommentDelegate.cs [hunk 3]
```diff
@@ -59,6 +59,19 @@ public DBResult<IEnumerable<Comment>> GetList(string hdId, string parentEntryId)
             return result;
         }
 
+        /// <inheritdoc />
+        public DBResult<IEnumerable<Comment>> GetAll(string hdId)
```
Comment (tiagograf-AOT) [inline Apps/Database/src/Delegates/DBCommentDelegate.cs:63]: this are change that are already in the dev branch, we need to update the features/9722 branch from dev so it doesn't display these changes and we can check the actual change you did!

File: Apps/WebClient/src/Server/Controllers/CommentController.cs [hunk 4]
```diff
@@ -139,11 +139,28 @@ public IActionResult Delete(string hdid, [FromBody] UserComment comment)
...
+        /// </summary>
+        /// <param name="hdid">The user hdid.</param>
+        /// <returns>The list of comments wrapped in a request result.</returns>
+        /// <response code="200">Returns the list of comments.</response>
+        /// <response code="401">The client must authenticate itself to get the requested response.</response>
+        /// <response code="403">The client does not have access rights to the content; that is, it is unauthorized, so the server is refusing to give the requested resource. Unlike 401, the client's identity is known to the server.</response>
+        [HttpGet]
+        [Route("{hdid}/[controller]")]
+        [Authorize(Policy = UserProfilePolicy.Read)]
+        public IActionResult GetAll(string hdid)
```
Comment (leobcgov) [inline Apps/WebClient/src/Server/Controllers/CommentController.cs:161]: Should we measure and update the unit tests as a part of this PR or create a new task for it as this new method will need to be tested?

File: Apps/Database/src/Delegates/DBCommentDelegate.cs (hunk 3 above)
Comment (leobcgov) [inline Apps/Database/src/Delegates/DBCommentDelegate.cs:63]: Unit tests for this method may need to be added if possible?

File: Apps/WebClient/src/Server/Services/CommentService.cs [hunk 5]
```diff
@@ -95,10 +95,47 @@ public RequestResult<IEnumerable<UserComment>> GetList(string hdId, string paren
...
+                PageIndex = 0,
+                PageSize = dbComments.Payload.Count(),
+                ResultStatus = dbComments.Status == DBStatusCode.Read ? ResultType.Success : ResultType.Error,
+                ResultError = dbComments.Status != DBStatusCode.Read ? new RequestResultError() { ResultMessage = dbComments.Message, ErrorCode = ErrorTranslator.ServiceError(ErrorType.CommunicationInternal, ServiceType.Database) } : null,
+            };
+            return result;
+        }
+
+        /// <inheritdoc />
+        public RequestResult<IDictionary<string, IEnumerable<UserComment>>> GetProfileComments(string hdId)
```
Comment (leobcgov) [inline Apps/WebClient/src/Server/Services/CommentService.cs:112]: we may need unit test for this new method as well.

File: Apps/WebClient/src/ClientApp/src/components/timeline/addComment.vue [hunk 6]
```diff
@@ -38,17 +42,22 @@ export default class AddCommentComponent extends Vue {
 
     private addComment(): void {
         this.isSaving = true;
-        this.commentService
-            .createComment(this.user.hdid, {
+        this.createComment({
+            hdid: this.user.hdid,
+            comment: {
+                id: "00000000-0000-0000-0000-000000000000",
```
Comment (leobcgov) [inline Apps/WebClient/src/ClientApp/src/components/timeline/addComment.vue:48]: aha nice :-)

**Review by leobcgov — CHANGES_REQUESTED** [review]
few questions for us to decide if we need to add unit tests as a part of this PR or not?

File: Apps/Database/src/Delegates/DBCommentDelegate.cs (hunk 3 above)
Comment (FuriousLlama) [reply Apps/Database/src/Delegates/DBCommentDelegate.cs:63]: yep, i did that, this is part of it

File: Apps/WebClient/src/Server/Controllers/CommentController.cs (hunk 4 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/Server/Controllers/CommentController.cs:161]: This is part of the merge from dev.

File: Apps/WebClient/src/Server/Services/CommentService.cs (hunk 5 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/Server/Services/CommentService.cs:112]: same as above

**Review by leobcgov — APPROVED** [review]

File: Apps/Database/src/Delegates/DBCommentDelegate.cs (hunk 3 above)
Comment (FuriousLlama) [reply Apps/Database/src/Delegates/DBCommentDelegate.cs:63]: same as above, merge from dev

**Review by tiagograf-AOT — APPROVED** [review]

**Pull request merged**